tripy/examples/segment-anything-model-v2/sam2/build_sam.py

```diff
@@ -0,0 +1,462 @@
+            "++model.sam_mask_decoder_extra_args.dynamic_multimask_stability_delta=0.05",
+            "++model.sam_mask_decoder_extra_args.dynamic_multimask_stability_thresh=0.98",
+            # the sigmoid mask logits on interacted frames with clicks in the memory encoder so that the encoded masks are exactly as what users see from clicking
+            "++model.binarize_mask_from_pts_for_mem_enc=true",
+            # fill small holes in the low-res masks up to `fill_hole_area` (before resizing them to the original video resolution)
+            "++model.fill_hole_area=8",
+        ]
+    hydra_overrides.extend(hydra_overrides_extra)
+
+    # Read config and init model
+    cfg = compose(config_name=config_file, overrides=hydra_overrides)
+    OmegaConf.resolve(cfg)
+    model = instantiate(cfg.model, _recursive_=True)
+    _load_checkpoint(model, ckpt_path, cfg)
+
+    current_dir = os.getcwd()
+    saved_engines_path = os.path.join(current_dir, "saved_engines")
+    # Create the saved_engines directory if it doesn't exist
+    if not os.path.exists(saved_engines_path):
+        os.makedirs(saved_engines_path)
+
+    # Get component configurations
+    components = get_component_configs(model, cfg)
+
+    for comp_name, comp_info in components.items():
+        if not comp_info["enabled"]:
+            continue
+
+        executable_file = os.path.join(saved_engines_path, comp_name)
+        if os.path.exists(executable_file):
+            print(f"Loading existing compiled {comp_name} from {executable_file}")
+            compiled_model = tp.Executable.load(executable_file)
+        else:
+            print(f"Compiling {comp_name}...")
+            start = time.time()
+            compiled_model = tp.compile(comp_info["model"], args=comp_info["compile_args"])
+            print(f"Compilation took {time.time() - start:.2f}s")
+            compiled_model.save(executable_file)
+
+        old_model = comp_info["model"]
+        # If model is model.forward, retrieve the original model object
+        if hasattr(old_model, "__self__"):
+            old_model = old_model.__self__
+
+        set_model_attr(model, comp_name, compiled_model)
+        if "special_handling" in comp_info and comp_info["special_handling"] is not None:
+            comp_info["special_handling"](old_model)
+
+    model = model.to(device)
+    if mode == "eval":
+        model.eval()
+    return model
+
+
+def load_component_weights(comp_name, component_info, state_dict, checkpoint_dict):
+    """
+    Load weights for a single component from checkpoint into state dict.
+    """
+
+    converted_keys = 0
+    for key in checkpoint_dict:
+        # Remove _true/_false suffixes if present
+        for suffix in ["_true", "_false", ".compiled_executable"]:
+            if comp_name.endswith(suffix):
+                comp_name = comp_name[: -len(suffix)]
+                break
+
+        if not key.startswith(comp_name) or (component_info.get("skip_load_state_dict") is True):
```

Comment:
The second condition doesn't seem to be key-related. We can add an early exit at the beginning of the function:
```py
converted_keys = 0

if component_info.get("skip_load_state_dict"):
    return converted_keys

for key in checkpoint_dict:
    ...
```